fluentui_controls/src/main/java/com/microsoft/fluentui/tokenized/controls/FileCard.kt

```diff
@@ -0,0 +1,136 @@
+import com.microsoft.fluentui.theme.token.Icon
+import com.microsoft.fluentui.theme.token.controlTokens.FileCardInfo
+import com.microsoft.fluentui.theme.token.controlTokens.FileCardTokens
+
+@Composable
+fun FileCard(
+    text: String,
+    subText: String,
+    modifier: Modifier = Modifier,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    onClick: (() -> Unit)? = null,
+    previewImageVector: ImageVector? = null,
+    @DrawableRes previewImageDrawable: Int? = null,
+    textIcon: FluentIcon,
+    actionOverflowIcon: FluentIcon? = null,
+    actionOverflowIconOnClick: (() -> Unit)? = null,
+    fileCardTokens: FileCardTokens? = null
+) {
+    val token = fileCardTokens
+        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.FileCard] as FileCardTokens
+    val isPreviewAvailable = !(previewImageDrawable == null && previewImageVector == null)
+    val fileCardInfo = FileCardInfo(isPreviewAvailable)
+    val textColor = token.textColor(fileCardInfo = fileCardInfo)
+    val subTextColor = token.subTextColor(fileCardInfo = fileCardInfo)
+    val iconColor = token.iconColor(fileCardInfo = fileCardInfo)
+    val actionOverFlowBackgroundColor =
+        token.actionOverFlowBackgroundColor(fileCardInfo = fileCardInfo)
+    val actionOverFlowIconColor = token.actionOverFlowIconColor(fileCardInfo = fileCardInfo)
+    val iconSize = token.iconSize(fileCardInfo = fileCardInfo)
+    val textTypography = token.textTypography(fileCardInfo = fileCardInfo)
+    val subTextTypography = token.subTextTypography(fileCardInfo = fileCardInfo)
+    val actionOverflowCornerRadius = token.actionOverflowCornerRadius(fileCardInfo = fileCardInfo)
+    val actionOverflowIconSize = token.actionOverflowIconSize(fileCardInfo = fileCardInfo)
+    val leadIconPadding = token.leadIconPadding(fileCardInfo = fileCardInfo)
+    val iconTextPadding = token.iconTextPadding(fileCardInfo = fileCardInfo)
+    val textSubTextPadding = token.textSubTextPadding(fileCardInfo = fileCardInfo)
+    val actionOverflowPadding = token.actionOverflowPadding(fileCardInfo = fileCardInfo)
+    val textVerticalPadding = token.textVerticalPadding(fileCardInfo = fileCardInfo)
+    BasicCard(
+        modifier = modifier,
+        onClick = onClick
+    ) {
+        Box(contentAlignment = Alignment.Center) {
+            Column(
+                modifier = Modifier.requiredWidth(IntrinsicSize.Min),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                if (previewImageDrawable != null) {
+                    Image(
+                        painterResource(id = previewImageDrawable), contentDescription = ""
+                    )
+                } else if (previewImageVector != null) {
+                    Image(
+                        imageVector = previewImageVector,
+                        contentDescription = ""
+                    )
+                }
+                Row(
```

Comment:
Lets consider FileCard in 2 part : Preview & Info. And take padding of infoSection as one token.